### PR TITLE
Story/form view model

### PR DESCRIPTION
**Pull request by laurens-ritense**
TP: https://ritense.tpondemand.com/entity/114204-fe-make-form-view-model-module

### CODE REVIEW

File: projects/valtimo/dossier/src/lib/services/dossier-tab.service.ts [hunk 1]
```diff
@@ -173,7 +173,7 @@ export class DossierTabService implements OnDestroy {
         );
       case ApiTabType.WIDGETS:
         return (
-          this.configService.featureToggles?.enableCaseWidgets &&
+          this.configService.config.featureToggles.enableCaseWidgets &&
```
Comment (mbritense) [inline projects/valtimo/dossier/src/lib/services/dossier-tab.service.ts]: rebuild config lib; revert

File: projects/valtimo/form-view-model/src/lib/form-view-model-routing.module.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import {ROLE_USER} from '@valtimo/config';
+import {AuthGuardService} from '@valtimo/security';
+import {FormViewModelComponent} from './components/form-view-model/form-view-model.component';
+
+const routes: Routes = [
+  {
+    path: '',
+    component: FormViewModelComponent,
+    canActivate: [AuthGuardService],
+    data: {title: 'FormViewModel', roles: [ROLE_USER], hidePageTitle: true},
```
Comment (mbritense) [inline projects/valtimo/form-view-model/src/lib/form-view-model-routing.module.ts]: remove, not used

File: projects/valtimo/form-view-model/src/lib/form-view-model-routing.module.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import {FormViewModelComponent} from './components/form-view-model/form-view-model.component';
+
+const routes: Routes = [];
+
+@NgModule({
+  declarations: [],
+  imports: [CommonModule, RouterModule.forChild(routes)],
+  exports: [RouterModule],
+})
+export class FormViewModelRoutingModule {}
```
Comment (mbritense) [inline projects/valtimo/form-view-model/src/lib/form-view-model-routing.module.ts]: remove, not used

File: projects/valtimo/form-view-model/src/lib/models/view-model.model.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export interface ViewModel {}
```
Comment (mbritense) [inline projects/valtimo/form-view-model/src/lib/models/view-model.model.ts]: what's. the point of an empty interface?

Comment (laurens-ritense) [reply projects/valtimo/form-view-model/src/lib/models/view-model.model.ts]: Teo said I should return an interface in the api service but the content from a ViewModel can't be predicted

Comment (mbritense) [reply projects/valtimo/form-view-model/src/lib/models/view-model.model.ts]: then use `object`

**Review by mbritense — DISMISSED** [review]

File: projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts [hunk 5]
```diff
@@ -204,14 +205,19 @@ export class DossierProcessStartModalComponent implements OnInit {
...
+    this.formViewModelDynamicContainer.clear();
     const formViewModelComponent = this.formViewModelDynamicContainer.createComponent(
         this.formViewModel.component
     );
     formViewModelComponent.instance.form = this.formDefinition;
     formViewModelComponent.instance.formName = this.formName;
     formViewModelComponent.instance.isStartForm = true;
     formViewModelComponent.instance.processDefinitionKey = this.processDefinitionKey;
+    formViewModelComponent.instance.documentDefinitionName = this.documentDefinitionName;
+    formViewModelComponent.instance.formSubmit.subscribe(() => {
```
Comment (mbritense) [inline projects/valtimo/dossier/src/lib/components/dossier-process-start-modal/dossier-process-start-modal.component.ts]: this subscription is not closed

**Review by mbritense — DISMISSED** [review]

**Review by mbritense — APPROVED** [review]

**Pull request merged**